src/Common/tests/System/PlatformDetection.Unix.cs

```diff
@@ -30,5 +30,10 @@ private static Version GetOSXKernelVersion()
 
         [DllImport("libc", SetLastError = true)]
         private static extern int sysctlbyname(string ctlName, byte[] oldp, ref IntPtr oldpLen, byte[] newp, IntPtr newpLen);
+
+        [DllImport("libc", SetLastError = true)]
+        internal static extern unsafe uint geteuid();
+
+        public static bool IsSuperUser => geteuid() == 0;
     }
-}
+}
```

Comment:
Thanks, I'll look out for that in future, but there's so many cases of this I'll not reset this job.